### DISCUSSION TITLE
De-Sync in multiplayer when using custom husk affliction. Works fine in Single player.

### DISCUSSION THREAD

**Discussion by phillipinman** [FakeFishGames/Barotrauma#14413] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

When using a custom husk affliction that transfers control to the player on death in a multiplayer game the game will desync for the affected player.

This does not occur when testing in the sub editor or single player.

While I can't say for sure, it seems this issue is introduced with the new re-spawn screen in the recent update. 

On disconnect an error related to the client being "discarded" is mentioned. I suspect the client reference to the character is thrown away once this re-spawn screen appears.

### Reproduction steps

1. Start a multiplayer game with two clients.
2. Kill the character to be infected.
3. Afflict the now dead character with a custom <huskaffliction> that has the "controlhusk="true" attribute.
4. wait for the character to be revived, once the character is "revived" and turned into a "husk" variant they de-sync as the game no longer seems to have the info about the client to give control back to the character.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server) and the Dedicated Server

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Linux and Windows

### Relevant error messages and crash reports

Unfortunately this issue isnt in the game logs so here is an extract from around the time I reproduced the issue:

```
753: Create Humanclownzombiesmart (6324, 6324)
756: Remove Human (6296, 6296)
```

Next we get a pop up desync error that says:
```
"Recieved msg: 2772, entity 6324 not found."
```
Link to the mod in question for debugging: https://gitlab.com/phillipinman/clown-o-trauma

NOTES:

Extract of the item in question:

```
  <!-- Cultist Smart Zombie Knife -->
  <Item identifier="clownotrauma_cultistknife_smartzombie" category="Equipment" Tags="smallitem,weapon,clownitem" cargocontaineridentifier="metalcrate" scale="0.5" impactsoundtag="impact_metal_light">
    <InventoryIcon name="Cultist Knife Smart Zombie" texture="Content/Items/InventoryIconAtlas.png" sourcerect="835,385,64,64" origin="0.5,0.5" />
    <Sprite name="Cultist Knife Smart Zombie Sprite" texture="Content/Items/Weapons/weapons_new.png" sourcerect="183,203,66,16" depth="0.55" origin="0.5,0.5" />
    <Body width="110" height="25" density="50" />
    <Fabricate suitablefabricators="clownotrauma_bloodaltar" requiredtime="40">
      <RequiredItem identifier="clownotrauma_cultistbloodknife" mincondition="0.99" maxcondition="0.99" usecondition="true" />
      <RequiredItem identifier="dementonite" />
    </Fabricate>
    <Deconstruct time="20" />
    <MeleeWeapon slots="Any,RightHand,LeftHand" aimpos="50,0" handle1="-10,0" holdangle="30" reload="0.75" range="50" msg="ItemMsgPickUpSelect">
      <Attack targetimpulse="10" severlimbsprobability="0.1" itemdamage="2">
        <Affliction identifier="lacerations" strength="2" />
        <Sound file="Content/Sounds/Damage/Gore1.ogg" range="500" />
      </Attack>
      <StatusEffect type="OnUse" target="This, UseTarget" condition="-100" comparison="and">
        <Conditional IsDead="true" />
        <Conditional ishuman="true" />
      </StatusEffect>
      <StatusEffect type="OnUse" target="UseTarget" setvalue="true" Isdead="false" comparison="and">
        <Conditional IsDead="true" />
        <Conditional ishuman="true" />
        <Affliction identifier="clownotrauma_resurrection" strength="100" />
        <ParticleEmitter particle="faradayfx" particlespersecond="4" particleburstinterval="2" particleburstamount="20" velocitymin="190" velocitymax="250" anglemin="0" anglemax="360" scalemin="0.4" scalemax="0.7" />
      </StatusEffect>
      <StatusEffect type="OnBroken" target="This">
        <Remove />
      </StatusEffect>
    </MeleeWeapon>
  </Item>
```

And the affliction it gives:

```  
<AfflictionHusk
    name="Clown Ressurection"
    identifier="clownotrauma_resurrection"
    description="HONK."
    type="alieninfection"
    targets="human"
    huskedspeciesname="[speciesname]clownzombiesmart"
    causeofdeathdescription="HONK!"
    selfcauseofdeathdescription="You have become one with the Honkmother."
    limbspecific="false"
    indicatorlimb="Torso"
    activationthreshold="0"
    showiconthreshold="40"
    showinhealthscannerthreshold="20"
    maxstrength="100"
    transferbuffs="true"
    dormantthreshold="50"
    activethreshold="75"
    transformthresholdondeath="75"
    transitionthreshold="100"
    controlhusk="true"
    iconcolors="60,140,195,255;60,107,195,255;60,0,195,255">
    <Effect minstrength="0" maxstrength="75"
      maxvitalitydecrease="0"
      strengthchange="0.3"
      minbodytint="195,195,195,0"
      maxbodytint="195,195,195,150"/>
    <Effect minstrength="75" maxstrength="100"
      maxvitalitydecrease="0"
      strengthchange="0.3"
      minbodytint="195,195,195,150"
      maxbodytint="195,195,195,200"/>
    <Icon texture="Content/UI/MainIconsAtlas.png" sourcerect="256,640,128,128" color="150,26,26,255" origin="0,0" />
  </AfflictionHusk>
```

P.S: Love the game, thanks for your hard work :)

**Comment by Regalis11** [comment]
Thank you for the report!

I gave this a test, but unfortunately couldn't reproduce the issue. In my test, the client gained control of the spawned "husk" with no issues:

https://github.com/user-attachments/assets/293159e3-4b14-4759-b7ad-e4c509acb6a4

The exact error messages (error message on disconnect, the error message about something being discarded) might help us diagnose the issue. Thanks and sorry for the trouble!

**Reply by phillipinman** [reply]
Hi, thanks for testing that, I'll try and get a video soon to showcase what I'm seeing on my end. I'll update this issue soon :)

**Comment by phillipinman** [comment]
Hopefully this helps: https://youtu.be/GDtlZHIMqXA

Naturally the other person is a human not a bot.

**Comment by phillipinman** [comment]
I think I found the issue, its self inflicted as always.

Doing the following on a character causes a desync in multiplayer and is 100 percent reproducible:

```
  <StatusEffect type="OnSpawn" target="This">
    <GiveTalentInfo talentidentifiers="clownotrauma_clownpower" />
  </StatusEffect>
```

Add the above to any creature and spawn it in, it will result in a desync.

I haven't tested but I'd guess any immediately applying status effect can do it.

As for how to fix the issue, it is entirely up to you. If you want to close this as "Won't Fix" I'd understand as under normal circumstances you wouldn't be making essentially a custom human-like monster with "HasInfo=True" and then adding a talent to it.

Adding "delay="3"" fixes the issue, I think it's only an issue on immediate spawn and applying a status effect.

The only issue is where to document this behavior as it won't show up in offline or in the sub editor.

Regardless many thanks for taking a look at the issue, I really appreciate it :)
